cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,130 @@
+					permissions:     mode.Perm().String(),
+					owner:           currentOwner.Username,
+					group:           currentGroup.Name,
+					hasCorrectOwner: currentOwner.Username == defaultPerm.owner,
+					hasCorrectGroup: currentGroup.Name == defaultPerm.group,
+				}
+
+				if mode.IsRegular() {
+					currentPerm.defaultPermissions = defaultPerm.file
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "f", fullPath, currentPerm, defaultPerm, verbose)
+				} else if mode.IsDir() {
+					currentPerm.defaultPermissions = defaultPerm.dir
+					currentPerm.hasCorrectPermissions = checkPermissions(currentPerm.permissions, currentPerm.defaultPermissions)
+					printOutput(level, "d", fullPath, currentPerm, defaultPerm, verbose)
+					FindRecursive(fullPath, defaultPerm, search, verbose, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions returns true if the permissions are correct (false in another case)
+/* currentPermissions [string] - permissions that the file or directory has (i.e. -rwxrwxr-x)
+ * defaultPermissions [string] - permissions that should have (i.e. rwxrwxr-x)
+ */
+func checkPermissions(currentPermissions, defaultPermissions string) bool {
+	return strings.Contains(currentPermissions, defaultPermissions)
+}
+
+// printOutput Prints the data in different formats according to the situation
+/* level [string] 														- used to print the output in a hierarchical way
+ * kind [string] 															- "f" if it is file or "d" in case of a directory
+ * fullPath [string] 				  								- file/directory full path (if verbose only use the name)
+ * currentPerm [currentPermissions]						- data to print relate
+ * 	currentPerm.permissions [string] 					- file/directory current permissions
+ * 	currentPerm.defaultPermissions [string] 	- file/directory default permissions
+ * 	currentPerm.owner [string] 								- file/directory current owner
+ * 	currentPerm.group [string] 								- file/directory current group
+ * 	currentPerm.hasCorrectPermissions [bool] 	- true if the permissions are correct
+ * 	currentPerm.hasCorrectOwner [bool] 				- true if the owner is correct
+ * 	currentPerm.hasCorrectGroup [bool] 				- true if the group is correct
+ * defaultPerm [defaultPermissions] 					- data regarding default permissions
+ * 	defaultPerm.file [string]									- default permissions for files
+ * 	defaultPerm.dir [string] 									- default permissions for directories
+ * 	defaultPerm.owner [string] 								- default owner
+ * 	defaultPerm.group [string] 								- default group
+ */
+func printOutput(level int, kind, fullPath string, currentPerm currentPermissions, defaultPerm defaultPermissions, verbose bool) {
+	if verbose {
+		fullPath = filepath.Base(fullPath)
+	}
+	hierarchy := strings.Repeat(" ", level)
+
+	if currentPerm.hasCorrectPermissions && currentPerm.hasCorrectOwner && currentPerm.hasCorrectGroup { // Everything correct
```

Comment:
hmm, please factor out common factors:

```
if currentPerm.hasCorrectPermissions {
 if ...
} else {
 if ...
}
```

yes, nested conditionals are hard to follow, but not because of the syntax, because of logic; by flattening it out the logic becomes even harder to follow.